infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -328,17 +339,45 @@ private void connectToClusterManager() {
      * - Building Log Replication Context (LR context shared across receiving and sending components)
      * - Start Log Replication Server (receiver component)
      */
-    private void bootstrapLogReplicationService() {
+    private void bootstrapLogReplicationService(ClusterRole role) {
         log.info("Bootstrap the Log Replication Service");
+
         // Through LogReplicationConfigAdapter retrieve system-specific configurations
         // such as streams to replicate and version
-        LogReplicationConfig logReplicationConfig = getLogReplicationConfiguration(getCorfuRuntime());
+        LogReplicationConfig logReplicationConfig =
+            getLogReplicationConfiguration(getCorfuRuntime());
+
+        Set<String> remoteClusterIds = new HashSet<>();
 
-        logReplicationMetadataManager = new LogReplicationMetadataManager(getCorfuRuntime(),
-            topologyDescriptor.getTopologyConfigId(), localClusterDescriptor.getClusterId());
+        if (role == ClusterRole.ACTIVE) {
+            remoteClusterIds.addAll(topologyDescriptor.getStandbyClusters()
+                .keySet());
+        } else if (role == ClusterRole.STANDBY) {
+            remoteClusterIds.addAll(topologyDescriptor.getActiveClusters()
+                .keySet());
+        } else {
+            log.debug("Cluster role is {}.  Not bootstrapping further.", role);
+            return;
+        }
+
+        for (String remoteClusterId : remoteClusterIds) {
+            LogReplicationMetadataManager metadataManager =
+                new LogReplicationMetadataManager(getCorfuRuntime(),
+                    topologyDescriptor.getTopologyConfigId(),
+                    remoteClusterId);
```

Comment:
I did not entirely look through all the code to be sure. But doesn't it make more sense, to have one metadata table (based on local node is as before) and instead of indexing by key (topologyConfigId/Version, etc.) index by remote cluster id.. anyways the value part is fixed and controlled by us ,so reading it all atomically shouldn't have important implications vs. reading one by one as before. Also, this table would have as many entries as remote clusters, which wouldn't be a scale issue, just one entry per remote.

Reply:
Yes definitely.

That is the behavior I added in the ReplicationStatus table in this PR.

We have another metadata table - CORFU-REPLICATION-WRITER-<local_cluster_id>.  It contains the topologyConfigId/Version etc as keys.  Ideally these should not have been the keys.  However, making the remote cluster id as the key will involve a schema change so I did not do it.

If we go by that approach, we may not need multiple metadata managers also.  As another minor change, I suggest removing the cluster_id suffix from the name too.

Reply:
There is also another issue... the way the key is currently used in ReplicationStatusTable is different on the active and standby.  Active uses the remote cluster id as key, standby uses local cluster id.

I changed all to use the remote cluster id in this PR.  So I think we will have to handle upgrade in any case.  There is value in changing the schema of  CORFU-REPLICATION-WRITER as you suggested.

Reply:
Resolving this comment as we have decided to overhaul the metadata tables schema and use a single metadata manager in a separate PR.